controllers/apply/form-router-next/lib/file-uploads.js

```diff
@@ -122,48 +110,80 @@ function uploadFile({ formId, applicationId, fileMetadata }) {
-        } else {
-            return { Key: 'av-status', Value: 'CLEAN' };
-        }
-    }).catch(err => {
-        logger.error(`Attempted read of unscanned file at ${fileMetadata.fileData.name}`, err);
-        throw new Error('ERR_FILE_SCAN_UNKNOWN');
+async function scanFile(filePath) {
+    logger.info(`Attempting virus scan for ${filePath}`);
+
+    const clamdscanConfig = {
+        socket: process.env.CLAMDSCAN_SOCKET || config.get('clamdscan.socket'),
+        timeout: config.get('clamdscan.timeout'),
+        local_fallback: true,
+        path: process.env.CLAMDSCAN_PATH || config.get('clamdscan.path'),
+        config_file:
+            process.env.CLAMDSCAN_CONFIG_FILE ||
+            config.get('clamdscan.config_file')
+    };
+
+    const clamscan = await new NodeClam().init({
+        debug_mode: isDev,
+        scan_recursively: false,
+        clamdscan: clamdscanConfig
     });
+
+    const { is_infected, viruses } = await clamscan.scan_file(filePath);
+
+    if (is_infected) {
+        logger.error(`Virus scan failed, file INFECTED`, { filePath, viruses });
+    } else {
+        logger.info(`Virus scan OK`, { filePath });
+    }
+
+    return { is_infected, viruses };
 }
 
 function checkAntiVirus({ formId, applicationId, filename }) {
     const keyName = [formId, applicationId, filename].join('/');
 
     return s3
-    .getObjectTagging({
-        Bucket: S3_UPLOAD_BUCKET,
-        Key: keyName
-    })
-    .promise()
-    .then(function(tags) {
-        const tagSet = get(tags, 'TagSet', []);
-        const avStatus = tagSet.find(t => t.Key === 'av-status');
-
-        if (avStatus && get(avStatus, 'Value') === 'CLEAN') {
-            return avStatus;
-        } else if (avStatus && get(avStatus, 'Value') === 'INFECTED') {
-            logger.error(`Infected file found at ${keyName}`);
-            throw new Error('ERR_FILE_SCAN_INFECTED');
-        } else {
-            logger.error(`Attempted read of unscanned file at ${keyName}`);
-            throw new Error('ERR_FILE_SCAN_UNKNOWN');
-        }
-    });
+        .getObjectTagging({
+            Bucket: S3_UPLOAD_BUCKET,
+            Key: keyName
+        })
+        .promise()
+        .then(function(tags) {
+            const tagSet = get(tags, 'TagSet', []);
+            const avStatus = tagSet.find(t => t.Key === 'av-status');
+
+            if (avStatus && get(avStatus, 'Value') === 'CLEAN') {
+                return avStatus;
+            } else if (avStatus && get(avStatus, 'Value') === 'INFECTED') {
+                logger.error(`Infected file found at ${keyName}`);
+                throw new Error('ERR_FILE_SCAN_INFECTED');
+            } else {
+                logger.error(`Attempted read of unscanned file at ${keyName}`);
+                throw new Error('ERR_FILE_SCAN_UNKNOWN');
+            }
+        });
+}
+
+function scanAndUpload({ formId, applicationId, fileMetadata }) {
+    function shouldScan() {
```

Comment:
Can we export/re-use this function for toggling the `await checkAntiVirus` in `index.js` or make this function usable within the `checkAntiVirus` function of `file-upload.js` ?

Reply:
The logic for that function is different so not sure we could reuse it

Reply:
I meant the `shouldScan` function.